examples_tests/common/CommonAPI.h

```diff
@@ -443,128 +681,256 @@ class CommonAPI
+
+		float queuePriority = 1.f;
+		constexpr uint32_t MaxQueueCount = 4;
+		video::ILogicalDevice::SQueueCreationParams qcp[MaxQueueCount] = {}; 
+		
+		uint32_t actualQueueCount = 1;
+		uint32_t mainQueueFamilyIndex = QueueFamilyProps::InvalidIndex;
+		if(graphicsQueueEnable)
+			mainQueueFamilyIndex = gpuInfo.queueFamilyProps.graphics.index;
+		else
+			mainQueueFamilyIndex = gpuInfo.queueFamilyProps.compute.index;
 
-			return getQueueFamilyIndex(gpu, requiredQueueFlags);
-		};
+		qcp[0].familyIndex = mainQueueFamilyIndex;
+		qcp[0].count = 1u;
+		qcp[0].flags = static_cast<video::IGPUQueue::E_CREATE_FLAGS>(0);
+		qcp[0].priorities = &queuePriority;
 
-		const auto familyIndex = getFamilyIndex();
+		if(qcp[0].familyIndex != gpuInfo.queueFamilyProps.compute.index)
+		{
+			qcp[actualQueueCount].flags = static_cast<video::IGPUQueue::E_CREATE_FLAGS>(0);
+			qcp[actualQueueCount].familyIndex = gpuInfo.queueFamilyProps.compute.index;
+			qcp[actualQueueCount].count = 1u;
+			qcp[actualQueueCount].priorities = &queuePriority;
+			actualQueueCount++;
+		}
+		if(gpuInfo.queueFamilyProps.transfer.index != gpuInfo.queueFamilyProps.compute.index && gpuInfo.queueFamilyProps.transfer.index != gpuInfo.queueFamilyProps.graphics.index)
+		{
+			qcp[actualQueueCount].flags = static_cast<video::IGPUQueue::E_CREATE_FLAGS>(0);
+			qcp[actualQueueCount].familyIndex = gpuInfo.queueFamilyProps.transfer.index;
+			qcp[actualQueueCount].count = 1u;
+			qcp[actualQueueCount].priorities = &queuePriority;
+			actualQueueCount++;
+		}
+		if(gpuInfo.queueFamilyProps.present.index != gpuInfo.queueFamilyProps.compute.index &&
+			gpuInfo.queueFamilyProps.present.index != gpuInfo.queueFamilyProps.graphics.index &&
+			gpuInfo.queueFamilyProps.present.index != gpuInfo.queueFamilyProps.transfer.index )
 		{
-			const bool status = result.surface->isSupportedForPhysicalDevice(gpu, familyIndex);
-			assert(status);
+			qcp[actualQueueCount].flags = static_cast<video::IGPUQueue::E_CREATE_FLAGS>(0);
+			qcp[actualQueueCount].familyIndex = gpuInfo.queueFamilyProps.present.index;
+			qcp[actualQueueCount].count = 1u;
+			qcp[actualQueueCount].priorities = &queuePriority;
+			actualQueueCount++;
 		}
 
 		video::ILogicalDevice::SCreationParams dev_params;
-		dev_params.queueParamsCount = 1u;
-		video::ILogicalDevice::SQueueCreationParams q_params;
-		q_params.familyIndex = familyIndex;
-		q_params.count = 1u;
-		q_params.flags = static_cast<video::IGPUQueue::E_CREATE_FLAGS>(0);
-		float priority = 1.f;
-		q_params.priorities = &priority;
-		dev_params.queueCreateInfos = &q_params;
+		dev_params.queueParamsCount = actualQueueCount;
+		dev_params.queueParams = qcp;
 		result.logicalDevice = gpu->createLogicalDevice(dev_params);
 
 		result.utilities = core::make_smart_refctd_ptr<video::IUtilities>(core::smart_refctd_ptr(result.logicalDevice));
 
-		auto queue = result.logicalDevice->getQueue(familyIndex, 0);
+		result.mainQueue = result.logicalDevice->getQueue(mainQueueFamilyIndex, 0);
```

Comment:
what do you need a `mainQueue` for?

Reply:
This is the queue that was used to create initOutput.commandPool and other stuff CommonAPI uses the queue for.

Reply:
Now that I think about it why should CommonAPI even create commandPool?

Reply:
ok we'll address it with your upcoming changes where you'll be creating up to 4 commandPools for the 4 queues (and always 4 cmdbuffers)

then user can just get the cmdbuffer for the Graphics/Compute queue as necessary